mayastor/src/grpc/mayastor/v1.rs

```diff
@@ -0,0 +1,196 @@
+use crate::{
+    core::Share,
+    grpc::{rpc_submit, GrpcClientContext, GrpcResult, Serializer},
+    lvs::{Error as LvsError, Lvs, PoolArgs},
+    subsys::PoolConfig,
+};
+use futures::FutureExt;
+use nix::errno::Errno;
+use std::{convert::TryFrom, fmt::Debug, time::Duration};
+use tonic::{Request, Response, Status};
+
+use rpc::mayastorv1::*;
+
+#[derive(Debug)]
+struct UnixStream(tokio::net::UnixStream);
+
+use ::function_name::named;
+use std::panic::AssertUnwindSafe;
+
+#[derive(Debug)]
+pub struct MayastorSvc {
+    name: String,
+    interval: Duration,
+    rw_lock: tokio::sync::RwLock<Option<GrpcClientContext>>,
+}
+
+#[async_trait::async_trait]
+impl<F, T> Serializer<F, T> for MayastorSvc
+where
+    T: Send + 'static,
+    F: core::future::Future<Output = Result<T, Status>> + Send + 'static,
+{
+    async fn locked(&self, ctx: GrpcClientContext, f: F) -> Result<T, Status> {
+        let mut guard = self.rw_lock.write().await;
```

Comment:
similarly here: I'd suggest re-using `client_context` (or simply `context`), or `context_guard` if you like.